File: Codigo/Frontend/src/app/interfaces/product.ts
export interface Product {
    code: string;
    name: string;
    description: string;
    price: number;
    selectedQuantity : number;
}

export class ProductClass {
  id: number;
  code: string;
  name: string;
  description: string;
  price: number;
  pharmacyId: number;
  selectedQuantity = 0;
  constructor(id: number, code: string, name: string, description: string, pharmacyId: number, price: number) {
    this.id = id;
    this.code = code;
    this.name = name;
    this.pharmacyId = pharmacyId;
    this.description = description;
    this.price = price;
  }

}

export class ProductDTOOut{
  id:number;
  code:string;
  name:string;
  description:string;
  price: number;

  constructor(id:number,code:string, name:string, description:string, price: number) {
    this.id = id;
    this.code = code;
    this.name = name;
    this.description = description;
    this.price = price;
  }
}

export class ProductDTOOutNew{
  code:string;
  name:string;
  description:string;
  price: number;
  selectedQuantity = 0;
  constructor(code:string, name:string, description:string, price: number) {
    this.code = code;
    this.name = name;
    this.description = description;
    this.price = price;
    
  }
}

File: Codigo/Backend/PharmaGo.Domain/Entities/Product.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;

namespace PharmaGo.Domain.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public Pharmacy? Pharmacy { get; set; }
        public decimal Price { get; set; }

        public Product()
        {
        }

        public void ValidOrFail()
        {
            if (string.IsNullOrWhiteSpace(Code))
            {
                throw new ValidationException("El código no puede ser vacío.");
            }
            /*
            if (productList.Any(p => p != this && p.Code == Code))
            {
                throw new ValidationException("El código ya está en uso por otro producto.");
            }*/

            if (string.IsNullOrWhiteSpace(Name))
            {
                throw new ValidationException("El nombre no puede ser vacío.");
            }

            if (string.IsNullOrWhiteSpace(Description))
            {
                throw new ValidationException("La descripción no puede estar vacía.");
            }
        }
    }
}

